demo/midgard-node/src/database/address.ts

```diff
@@ -0,0 +1,124 @@
+import { Database } from "@/services/database.js";
+import { SqlClient, SqlError } from "@effect/sql";
+import { Effect } from "effect";
+import { mapSqlError } from "@/database/utils/common.js"
+import { Address, CML } from "@lucid-evolution/lucid";
+import * as MempoolDB from "@/database/mempool.js"
+import * as ImmutableDB from "@/database/immutable.js"
+import * as Tx from "@/database/utils/tx.js"
+import * as Ledger from "@/database/utils/ledger.js"
+import { MempoolLedgerDB } from "./index.js";
+
+const tableName = "address"
+
+export type Entry = {
+  [Ledger.Columns.TX_ID]: Buffer;
+  [Ledger.Columns.ADDRESS]: Address;
+};
+
+export const createTable = (
+): Effect.Effect<void, Error, Database> =>
+  Effect.gen(function* () {
+    const sql = yield* SqlClient.SqlClient;
+    yield* sql`CREATE TABLE IF NOT EXISTS ${sql(tableName)} (
+      ${sql(Ledger.Columns.TX_ID)} BYTEA NOT NULL,
+      ${sql(Ledger.Columns.ADDRESS)} BYTEA NOT NULL,
+    );`;
+  }).pipe(Effect.withLogSpan(`creating table ${tableName}`), mapSqlError);
+
+
+export const insert = (
+  spent: Buffer[],
+  produced: Ledger.Entry[]
+): Effect.Effect<void, Error, Database> =>
+  Effect.gen(function* () {
+    yield* Effect.logDebug(`${tableName} db: attempt to insert entries`);
+    const sql = yield* SqlClient.SqlClient;
+
+    const inputEntries = yield* sql<Entry>
+    `SELECT (${sql(Ledger.Columns.TX_ID)}, ${sql(Ledger.Columns.ADDRESS)})
+    FROM ${sql(MempoolLedgerDB.tableName)}
+    WHERE ${sql(Ledger.Columns.TX_ID)} IN ${sql.in(spent)}`;
+
+    const outputEntries : Entry[] = produced.map((e) => ({
+      [Ledger.Columns.TX_ID]: e[Ledger.Columns.TX_ID],
+      [Ledger.Columns.ADDRESS]: e[Ledger.Columns.ADDRESS]
+    }))
+
+    yield* sql`INSERT INTO ${sql(tableName)} ${sql.insert([...inputEntries, ...outputEntries])}`;
```

Comment:
You can reuse your `insertEntries` here :)
```suggestion
    yield* insertEntries([...inputEntries, ...outputEntries]);
```